### ISSUE TITLE
Requesting Pulsar to support IoT protocols - STOMP, AMQP, MQTT, WSS

### ISSUE THREAD

**Issue by PrashantKS** [apache/pulsar#6587]
Use case:
In the IoT related use cases, we are using RabbitMQ as the message broker to receive/send messages from IoT devices over the protocols STOMP, AMQP, MQTT, WSS and further integrate with Kafka/Pulsar for streaming processing/data analysis. 

Request:
It would be better if Pulsar has a proxy handler for the protocols STOMP, AMQP, MQTT, WSS. It may help millions of projects, especially those that are serving IoT use cases.

**Comment by sijie** [comment]
Hi @PrashantKS thank you for creating the issue. 

We have just added kafka protocol support to pulsar via KoP (https://github.com/streamnative/kop). It is

We are also working on adding both AMQP and MQTT to Pulsar through the same protocol handler mechanism. Please stay tuned. It is coming out soon.

**Comment by davidlanouette** [comment]
My team also is seriously considering Pulsar to replace ActiveMQ 5 (yes, it's way overdue), but we have a lot of STOMP and AMQP clients.

@sijie Does the `kop` project depend on the streamio platform, or will it run on native Pulsar?

**Comment by sijie** [comment]
@davidlanouette No. `kop` doesn't depend on the streamnative platform. `kop` is a protocol handler that you can install it natively in your existing Pulsar cluster. Also streamnative platform is just an enterprise version of pulsar with addons.

We are about to release AMQP-on-Pulsar (AoP) and MQTT-on-Pulsar (MoP) in a few weeks.

**Comment by davidlanouette** [comment]
@sijie Thanks for the update!  

Do you have a link to that repo?  I'd be interested in contributing if I can.

**Comment by vzhikserg** [comment]
This issue contains too many requests in one. I would suggest to create 4 separate issues (one for each protocol)  and close this one.  The issues can be closed as soon as the handlers are implemented and we don't have to wait till all other handlers are finished.

**Comment by sijie** [comment]
@davidlanouette sorry that I missed this message!

The AoP is available at https://github.com/streamnative/aop. MoP will be also public soon.

@vzhikserg I agree with you. We can create separate issues for the requested protocol handlers and then close this one.

**Comment by lloydchandran-zafinlabs** [comment]
> @davidlanouette sorry that I missed this message!
> 
> The AoP is available at https://github.com/streamnative/aop. MoP will be also public soon.
> 
> @vzhikserg I agree with you. We can create separate issues for the requested protocol handlers and then close this one.

@sijie  Are you guys planning to add support for protocols like STOMP ?.  Is there a different issue created for this ?

**Comment by jaggerwang** [comment]
> > @davidlanouette sorry that I missed this message!
> > The AoP is available at https://github.com/streamnative/aop. MoP will be also public soon.
> > @vzhikserg I agree with you. We can create separate issues for the requested protocol handlers and then close this one.
> 
> @sijie Are you guys planning to add support for protocols like STOMP ?. Is there a different issue created for this ?

Same request, I want to use pulsar as my spring websocket server's STOMP message broker.

**Comment by hpvd** [comment]
MQTT on Pulsar (MoP) can be found here:
https://github.com/streamnative/mop
